src/core/lib/channel/metrics.h

```diff
@@ -317,13 +318,16 @@ class GlobalStatsPluginRegistry {
 
     // Registers a callback to be used to populate callback metrics.
     // The callback will update the specified metrics.  The callback
-    // will be invoked no more often than min_interval.
+    // will be invoked no more often than min_interval.  Multiple callbacks may
+    // be registered for the same metrics, as long as no two callbacks report
+    // data for the same set of labels in which case the behavior is undefined.
```

Comment:
I think we should define the behavior. This is not public API so it's not as if the wiggle room is important. Let's just say we will report both values and let OTel handle it.

Reply:
Sorry I forgot to mention another scenario (I'm always confused about this). Within the same registered callback, if we decide to update the cache, it looks like the last value with the same label_set will be recorded, but there could be multiple registered callbacks for the same instrument, say we are still within the same gauge interval and another callback fired later but decided that it did not want to update the cache because it has a larger timeout, so it would still use the cached value and overwrite the updated value from the previous registered callback.

Because of this complication, I think it's probably better to say the behavior is undefined.

Reply:
I agree, better to just say it's undefined.  As Yijie mentioned, it's actually quite hard to explain which value will win.  But more fundamentally, we really don't want callers to do this in the first place, so let's not invite a Hyrum's Law violation by defining the behavior.